### PR TITLE
adding the ability to just add a config object is stead of a file location.

### PR DESCRIPTION
**Pull request by mientjan**
adding the ability to just add a config object instead of a file location.

### CODE REVIEW

File: src/lib/Uploader.ts [hunk 1]
```diff
@@ -46,7 +47,11 @@ export default class Uploader {
     };
 
     if (this.options.config) {
-      AWS.config.loadFromPath(this.options.config);
+      if (typeof this.options.config === 'string') {
+        AWS.config.loadFromPath(this.options.config);
+      } else if (typeof this.options.config === 'object') {
+        AWS.config.constructor(this.options.config);
```
Comment (ThaNarie) [inline src/lib/Uploader.ts:53]: Shouldn't we be using the actual constructor?

```suggestion
        new AWS.config(this.options.config);
```

File: src/lib/Uploader.ts [hunk 2]
```diff
@@ -46,7 +47,11 @@ export default class Uploader {
...
+      }
```
Comment (ThaNarie) [inline src/lib/Uploader.ts:54]: Can we add an `else` with an error that an unsupported Config object is passed, so not used?

**Review by ThaNarie — CHANGES_REQUESTED** [review]
Besides the 2 comments, could you also update the documentation to reflect this addition? :)

File: src/lib/Uploader.ts (hunk 1 above)
Comment (mientjan) [reply src/lib/Uploader.ts:53]: Where is it stored then? this way it works exactly like AWS.config.loadFromPath().

File: src/lib/Uploader.ts (hunk 2 above)
Comment (mientjan) [reply src/lib/Uploader.ts:56]: done

File: src/lib/Uploader.ts (hunk 1 above)
Comment (ThaNarie) [reply src/lib/Uploader.ts:53]: > Where is it stored then?
Nowhere, but it doesn't have to. `AWS.config.loadFromPath` also returns the newly created Config object that isn't stored anywhere.

> this way it works exactly like AWS.config.loadFromPath()
Not *exactly* :)

`loadFromPath` loads config from path, and returns a new Config object.

Both `new AWS.Config` and `AWS.config.constructor` just call the constructor, but the latter uses a "dynamic reference" to that constructor. In my opinion it's not needed to use that dynamic nature while you could just use the normal `Config` constructor itself.

Comment (mientjan) [reply src/lib/Uploader.ts:53]: > AWS.config.loadFromPath 
does not create a new config instance?

> AWS.config.constructor
i'm recalling constructor so i can set a object.

AWS.config is a static property.

Comment (ThaNarie) [reply src/lib/Uploader.ts:53]: Okay, to summarise:

We're calling the `constructor` of the existing `config` instance as a "normal function" to internally update the settings we pass to it.

Note that this method is not supported when ES6 classes would be used internally, but at this point is fine.

There is a rewrite of the ASW library going on, but that would most definitely require other updates to this library to make it work again, so we've decided to keep it in like this.

**Review by ThaNarie — APPROVED** [review]

**Comment by ThaNarie** [comment]
Published to NPM as `1.5.0`

**Pull request merged**